### PR TITLE
Fix song continuously rewinding when seeking using keyboard arrow keys.

### PR DESCRIPTION
**Pull request by eduardosm**

### CODE REVIEW

File: src/engines/gstenginepipeline.h [hunk 1]
```diff
@@ -258,6 +258,13 @@ class GstEnginePipeline : public QObject {
   bool pipeline_is_connected_;
   qint64 pending_seek_nanosec_;
 
+  // We can only use gst_element_query_position() when the pipeline is in
+  // PAUSED nor PLAYING state. Whenever we get a new position (e.g. after a
+  // correct call to gst_element_query_position() or after a seek), we store
+  // it here so that we can use it when using gst_element_query_position() is
+  // not possible.
+  mutable gint64 last_known_position_;
```
Comment (hatstand) [inline src/engines/gstenginepipeline.h]: Change to `last_known_position_ns_`

**Comment by eduardosm** [comment]
Done

File: src/engines/gstenginepipeline.h [hunk 2]
```diff
@@ -258,6 +258,13 @@ class GstEnginePipeline : public QObject {
   bool pipeline_is_connected_;
   qint64 pending_seek_nanosec_;
 
+  // We can only use gst_element_query_position() when the pipeline is in
+  // PAUSED nor PLAYING state. Whenever we get a new position (e.g. after a
+  // correct call to gst_element_query_position() or after a seek), we store
+  // it here so that we can use it when using gst_element_query_position() is
+  // not possible.
+  mutable gint64 last_known_position_ns_;
```
Comment (ArnaudBienner) [inline src/engines/gstenginepipeline.h]: Why is this mutable?

Comment (eduardosm) [inline src/engines/gstenginepipeline.h]: Because GstEnginePipeline::position(), which is const, needs to modify this value.

Comment (ArnaudBienner) [inline src/engines/gstenginepipeline.h]: oh OK, I missed this. Nevermind then.

**Pull request merged**